### ISSUE TITLE
Incorrect checkerboard pattern detection

### ISSUE THREAD

**Issue by shrutarv** [ethz-asl/kalibr#608]
I have used multiple combinations of checkerboard patterns for calibration. But, kalibr does not detect the corners in an ordered way. please have a look at the attached image. Is this just a visualization error? Please help me resolve this.
![issue](https://user-images.githubusercontent.com/49610063/231991841-03030eeb-cb73-45e1-a8fa-7286572f4275.png)
![Screenshot from 2023-04-14 10-37-28](https://user-images.githubusercontent.com/49610063/231991847-c7b2f5db-fef2-4ead-b21f-648cbcd9c0a4.png)

**Comment by goldbattle** [comment]
No, this is the problem of using checkerboards (it isn't clear the ordering so you can have it "flip" mid way through a dataset). This is also why it is not possible to have a square checkerboard. See the top of the wiki page:
https://github.com/ethz-asl/kalibr/wiki/calibration-targets

**Comment by shrutarv** [comment]
I tried with april grid. But, kalibr does not detect the grid patterns. Please have a look at the screenshot
![detection_issue](https://user-images.githubusercontent.com/49610063/232087620-886c8972-0b69-4e92-99ed-deaad2af10ab.png)

**Comment by goldbattle** [comment]
Is this a thermal camera? The contrast seems very low. Likely it is a resolution or focus problem with your sensor.

**Comment by shrutarv** [comment]
> No, this is the problem of using checkerboards (it isn't clear the ordering so you can have it "flip" mid way through a dataset). This is also why it is not possible to have a square checkerboard

I also tried with a 6x9 checkerboard pattern. Please have a look at the second image. It looks like there is a bug in kalibr. Could you please hep me out here.

**Comment by goldbattle** [comment]
The detection of the checkerboards look ok in your original posting. What is the problem or could you clarify?

See the example here:
https://docs.opencv.org/3.4/d4/d94/tutorial_camera_calibration.html
![image](https://user-images.githubusercontent.com/2222562/232558175-f54c8d87-d6ec-4493-8493-3d71ceba32d3.png)


Maybe you are referring to the colors not being per-row in your picture? This maybe is a bug in visualization or how the extractions are tracked. Is this what you mean / could you clarify? I originally thought you were talking about that detection can be on the horizontal rows or the vertical columns.

**Comment by goldbattle** [comment]
Very good point, should be looked into.
